### PR TITLE
002 create employee

### PR DESCRIPTION
**Pull request by RaggedyRach**
please review create an employee, with sql and REST function

### CODE REVIEW

**Comment by RaggedyRach** [comment]
all checks have finally passed

File: src/main/java/org/example/TestApplication.java [hunk 1]
```diff
@@ -32,6 +35,9 @@ public void run(final TestConfiguration configuration,
                     final Environment environment) {
         environment.jersey()
```
Comment (boydstepkainos) [inline src/main/java/org/example/TestApplication.java:36]: We can delete this test one

File: src/main/java/org/example/daos/EmployeeDao.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.ResultSet;
+import java.sql.PreparedStatement;
+import java.util.ArrayList;
+import java.util.List;
+
+public class EmployeeDao {
+
+    public List<Employee> getAllEmployees() throws SQLException {
```
Comment (boydstepkainos) [inline src/main/java/org/example/daos/EmployeeDao.java:15]: This function doesn't look to be used as part of this user story/requirement. Lets remove it for now

File: src/main/java/org/example/daos/EmployeeDao.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                employees.add(order);
+            }
+        }
+        return employees;
+    }
+
+    public int createEmployee(final EmployeeRequest employeeRequest)
+            throws SQLException {
+
+        Connection connection = DatabaseConnector.getConnection();
```
Comment (boydstepkainos) [inline src/main/java/org/example/daos/EmployeeDao.java:48]: Some error handling needs added here and lets close the connection in a finally block

File: src/main/java/org/example/exceptions/Entity.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package org.example.exceptions;
+
+public enum Entity {
+    EMPLOYEE("Employee"),
+    SALES_EMPLOYEE("SalesEmployee"),
```
Comment (boydstepkainos) [inline src/main/java/org/example/exceptions/Entity.java:5]: Parts of this code isn't needed until later stories, one of the non functional requirements is that each PR contains only the code necessary for the functionality in the story.

**Review by boydstepkainos — CHANGES_REQUESTED** [review]

File: src/main/java/org/example/exceptions/Entity.java (hunk 4 above)
Comment (RaggedyRach) [reply src/main/java/org/example/exceptions/Entity.java:5]: Thanks, is it ok to comment out? or delete?

File: src/main/java/org/example/daos/EmployeeDao.java (hunk 2 above)
Comment (RaggedyRach) [reply src/main/java/org/example/daos/EmployeeDao.java:15]: commented out. did first by error

File: src/main/java/org/example/daos/EmployeeDao.java (hunk 3 above)
Comment (RaggedyRach) [reply src/main/java/org/example/daos/EmployeeDao.java:48]: put the connectio in a try block, which will close it. thx

File: src/main/java/org/example/exceptions/Entity.java (hunk 4 above)
Comment (RaggedyRach) [reply src/main/java/org/example/exceptions/Entity.java:5]: commented out other entities not needed

**Review by RaggedyRach — COMMENTED** [review]
went through comments and made changes. If you could please review. thanks

File: src/main/java/org/example/TestApplication.java (hunk 1 above)
Comment (RaggedyRach) [reply src/main/java/org/example/TestApplication.java:36]: think deleted

**Comment by RaggedyRach** [comment]
changes to review and passed style and run checks

File: src/main/java/org/example/controllers/EmployeeController.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+     EmployeeService employeeService;
+
+    public EmployeeController(
+            final EmployeeService employeeService) {
+            this.employeeService = employeeService;
+    }
+
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response createEmployee(final EmployeeRequest employeeRequest) {
```
Comment (boydstepkainos) [inline src/main/java/org/example/controllers/EmployeeController.java:26]: Requirements asks for a delivery employee but the naming here is generic employee

Comment (RaggedyRach) [reply src/main/java/org/example/controllers/EmployeeController.java:26]: Dyln has delivery employee that extends the employee class and the delivery employee table

File: src/main/java/org/example/exceptions/Entity.java (hunk 4 above)
Comment (boydstepkainos) [reply src/main/java/org/example/exceptions/Entity.java]: Lets delete, commented out code doesn't get merged in real projects

File: src/main/java/org/example/models/EmployeeResponse.java [hunk 6]
```diff
@@ -0,0 +1,5 @@
+package org.example.models;
```
[the remark this answers is not in the document]
Comment (RaggedyRach) [reply src/main/java/org/example/models/EmployeeResponse.java:1]: ok these have been deleted

File: src/main/java/org/example/exceptions/Entity.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
+package org.example.exceptions;
+
+public enum Entity {
+    EMPLOYEE("Employee");
+//    SALES_EMPLOYEE("SalesEmployee"),
```
[the remark this answers is not in the document]
Comment (RaggedyRach) [reply src/main/java/org/example/exceptions/Entity.java:5]: these have been removed, thanks

File: src/main/java/org/example/daos/EmployeeDao.java [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.ResultSet;
+import java.sql.PreparedStatement;
+//import java.util.ArrayList;
+//import java.util.List;
+
+public class EmployeeDao {
+
+//    public List<Employee> getAllEmployees() throws SQLException {
```
[the remark this answers is not in the document]
Comment (RaggedyRach) [reply src/main/java/org/example/daos/EmployeeDao.java:17]: removed commented code, thanks!

**Comment by RaggedyRach** [comment]
all commented code deleted

**Comment by ntotten-kainos** [comment]
Merger conflicts resolved

**Review by bbell-kainos — APPROVED** [review]

**Pull request merged**